Implement search and filter for Integrations tab

Lukewh opened the pull request:
## Done
- Implemented search and filter, which meant:
  - Introduce [RecoilJS](https://recoiljs.org) for state management
  - Introduce ReactQuery for query calls
  - Add a bunch of state
  - Do a bunch of refactoring

## How to QA
- Visit the demo and view all your favourite charms. Some of my favourite: Openstack Dashboard, Swift Storage, and postgresql
- Click on the integrations tab
- Filter the cards with the available filter
- Click around

## Issue / Card
Fixes https://warthogs.atlassian.net/browse/WD-510

## Screenshots
![image](https://user-images.githubusercontent.com/479384/217946449-6a2ea740-9287-4548-a5ab-0c8260d7bfdc.png)

webteam-app commented:
Demo starting at https://charmhub-io-1485.demos.haus

steverydz commented on package.json:
```diff
@@ -54,7 +54,9 @@
     "react-dom": "18.2.0",
     "react-dropzone": "^14.2.2",
     "react-markdown": "8.0.4",
+    "react-query": "^3.39.2",
```
Pin version please

steverydz commented on package.json:
```diff
@@ -54,7 +54,9 @@
     "react-markdown": "8.0.4",
+    "react-query": "^3.39.2",
     "react-router-dom": "6.6.2",
+    "recoil": "^0.7.6",
```
Pin version please

steverydz commented on package.json:
```diff
@@ -54,7 +54,9 @@
     "react-dom": "18.2.0",
     "react-dropzone": "^14.2.2",
```
Not your fault, but pin version please

steverydz reviewed (approved):
LGTM 👍 

Just pin those new dependency versions and all good

anasereijo commented:
- [ ] the padding between the title and the _hr_ gets bigger if we select a lot of chips from the search dropdown. The _hr_ also moves up and down.  The size of the dropdown shouldn't affect the page.

https://user-images.githubusercontent.com/36884067/218431695-7b1c2eec-3265-4d8c-882d-4fd2d001224b.mov

- [ ] dropdown: Integrations should be singular as per all the other items
<img width="370" alt="Screenshot 2023-02-13 at 10 21 56" src="https://user-images.githubusercontent.com/36884067/218432506-5a2f5a34-d4e3-43ac-b47b-d100e604f2f3.png">

- [ ] cards don't look like the[ design provided](https://zpl.io/jZrDmZ4)

- [ ] Description is missing from each relation

- [ ] interfaces links missing from the headings
final design
![Screenshot 2023-02-13](https://user-images.githubusercontent.com/36884067/218434993-d1977f74-c1a1-49b2-a958-59c400434ac6.jpg)
how it looks like
![Screenshot 2023-02-13 at 10 34 33](https://user-images.githubusercontent.com/36884067/218435213-f28768ef-155b-4ec2-a9ee-3218125f739d.jpg)

- [ ] can we limit the height of the cards to be always the same? maybe cut the name if too big
![Screenshot 2023-02-13](https://user-images.githubusercontent.com/36884067/218435711-c560a951-4369-45ef-b930-d8402631efc6.jpg)

- [ ] can we make the whole card clickable as per homepage?

Lukewh commented:
> * [ ]  Description is missing from each relation

Relations don't have descriptions, unless they're a new “canonical interface”

> * [ ]  interfaces links missing from the headings

Where would they link to? There's no page for an interface, unless it's a “canonical interface”

Lukewh commented:
> * [x]  cards don't look like the[ design provided](https://zpl.io/jZrDmZ4)

@anasereijo I've addressed most of the issues, however, “Recommended by publisher” doesn't currently exist and adding the libraries is an extra API call that will increase loading times and, in my opinion, probably isn't worth it right now. Wdyt?

anasereijo commented:
-1 lets chat tomorrow
